### ISSUE TITLE
Memory Requirements for TREC-CAR set

### ISSUE THREAD

**Issue by candeveci13** [nyu-dl/dl4marco-bert#49]
Hi,

I'm trying to reproduce your results for TREC-CAR dataset. When I try to execute run_treccar.py with the default parameters I get the error reading "Resource exhausted: OOM when allocating tensor with shape[16384,1024]". I reduced the batch size down to 2 and could be able to train/evaluate. Naturally, I obtained lower performance results with this configuration. Could you specify the memory requirement for this part? And/or could you please share the specifics of the hardware you used?

What I used was a Tesla V100 GPU with 16GB of RAM. 

Thanks in advance,

**Comment by rodrigonogueira4** [comment]
Sorry for the late reply. This model was trained on a Google TPUs v3, which has 128GB. To train a BERT-large on a 16GB GPU you will need to use gradient accumulation. Otherwise, you would have to use small batches (e.g. 2), and this would result in a model with poor effectiveness.
